core/cache/package_test.go

```diff
@@ -66,6 +67,86 @@ func (s *BaseSuite) NewHub() *pubsub.SimpleHub {
 	return pubsub.NewSimpleHub(&pubsub.SimpleHubConfig{Logger: logger})
 }
 
+func (s *BaseSuite) New(c *gc.C) (*Controller, <-chan interface{}) {
+	events := s.CaptureEvents(c)
```

Comment:
Could we simplify ProcessChange by putting events as a member of BaseSuite instead of setting it in the child's constructor and then passing it in each time?